### PR TITLE
refactor: elm rebuild

### PR DESCRIPTION
**Pull request by plyr4**
this PR includes all of the changes mentioned in [PR #752](https://github.com/go-vela/ui/pull/752):

refactoring the code to follow established patterns better, inspired by [elm-land](https://elm.land/concepts/) and [NextJS](https://nextjs.org/)

the main idea is to build the app around the following core components
- Shared
- Layouts
- Pages
- View
- Components
- Auth

some other welcome changes that are included:
- `/add-deployment` is now `/deployments/add` (consistency) 
- `/add-schedule` is now `/schedules/add` (consistency)
- `/deployments/add` supports query parameters

### CODE REVIEW

**Review by ecrupper — DISMISSED** [review]
🤯

File: src/elm/Pages/Org_/Repo_/Build_.elm [hunk 1]
```diff
@@ -0,0 +1,744 @@
...
+            )
+
+        -- STEPS
+        GetBuildStepsResponse options response ->
+            case response of
+                Ok ( _, steps ) ->
+                    ( { model | steps = RemoteData.succeed steps }
+                    , steps
+                        |> List.Extra.find (\step -> Maybe.withDefault -1 model.focus.group == step.number)
+                        |> Maybe.map (\step -> step)
```
Comment (wass3rw3rk) [inline src/elm/Pages/Org_/Repo_/Build_.elm:278]: is this needed?

File: tests/NoneTest.elm
Comment (wass3rw3rk) [inline tests/NoneTest.elm:1]: should we just remove this? it's doing even less than what we had before 😓

**Review by wass3rw3rk — APPROVED** [review]
🥳 a new beginning! mammoth effort and nice work! GitHub UI diff view choked a little bit here and there 😓

i did point some things out, but happy to address in a follow-up.

File: src/elm/Pages/Org_.elm [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+
+
+view : Shared.Model -> Route { org : String } -> Model -> View Msg
+view shared route model =
+    { title = "Repos" ++ Util.pageToString (Dict.get "page" route.query)
+    , body =
+        [ Html.caption
+            [ class "builds-caption"
+            ]
+            [ Html.span [] []
```
[the remark this answers is not in the document]
Comment (plyr4) [reply src/elm/Pages/Org_.elm:191]: good catch, ill do another pass on `Html.*` and remove them

File: src/elm/Pages/Org_/Repo_/Build_.elm (hunk 1 above)
Comment (plyr4) [reply src/elm/Pages/Org_/Repo_/Build_.elm:278]: good catch, nope!

File: tests/NoneTest.elm
Comment (plyr4) [reply tests/NoneTest.elm:1]: iirc elm-test bombs with no tests, so `NoneTest.elm` was a no-op to keep it happy. 

i'll look into it, but will likely open a new issue to do it correctly

File: src/elm/Main.elm [hunk 3]
```diff
@@ -5,5188 +5,3037 @@ SPDX-License-Identifier: Apache-2.0
...
-        DownloadFile ext fn filename content ->
-            ( model
-            , Download.string filename ext <| fn content
+                ( defaultLayoutModel, defaultLayoutEffect ) =
+                    Layout.init defaultLayout ()
+            in
+            ( Main.Layouts.Model.Default_Org { default = defaultLayoutModel, org = nestedLayoutModel }
+            , Cmd.batch
+                [ fromLayoutEffect model (Effect.map Main.Layouts.Msg.Default_Org nestedLayoutEffect)
+                , fromLayoutEffect model (Effect.map Main.Layouts.Msg.Default defaultLayoutEffect)
```
[the remark this answers is not in the document]
Comment (plyr4) [reply src/elm/Main.elm:189]: i believe order doesnt matter, i didnt see any issues while testing. 
though to be fair this is elm.land boilerplate code, i recommend we don't modify it

File: tests/NoneTest.elm
Comment (plyr4) [reply tests/NoneTest.elm:1]: but to your point the old route tests were legit, i pulled that back in and modified it to work with the `Route.Path` elm.land module that replaced `Routes`

**Review by ecrupper — DISMISSED** [review]

**Review by wass3rw3rk — DISMISSED** [review]

**Review by ecrupper — APPROVED** [review]

**Review by KellyMerrick — APPROVED** [review]
lgtm

**Pull request merged**